app/models/assignment_repo/creator.rb

```diff
@@ -189,5 +190,69 @@ def suffixed_repo_name(repository_name, suffix_count)
       suffix = "-#{suffix_count}"
       repository_name.truncate(100 - suffix.length, omission: '') + suffix
     end
+
+    # Internal: Handles assignment repository setup
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    #
+    # Returns nothing
+    def setup_repository(configs_repo, assignment_repo)
+      return unless configs_repo.branch_present? 'github-classroom'
+      configs_tree = configs_repo.branch_tree('github-classroom')
+      configs_tree.tree.each do |configuration|
+        process_configuration(configs_repo, assignment_repo, configuration)
+      end
+
+      wait_import_completion assignment_repo
+
+      assignment_repo.remove_branch('github-classroom')
+    end
+
+    # Internal: Wait for import completion
+    #
+    # github_repository - GitHubRepository awaiting import completion
+    #
+    # Returns nothing
+    def wait_import_completion(github_repository)
+      wait_limit = 60
+      increment_rate = 0.25
+      t = 1
+      loop do
+        break if github_repository.import_progress[:status] == 'complete' || t > wait_limit
```

Comment:
We'll need error handling here as well. import_progress can return an exception.